### PR TITLE
Update UI package to latest version

### PR DESCRIPTION
**Pull request by AllForNothing**
Fixes: #13137 (Upgrade Angular to 10, Upgrade Clarity to 4)
Temp Env: https://10.185.12.35
1. All unit tests passed
2. All nightly tests passed
3. Manully tested for all pages 

Signed-off-by: AllForNothing <sshijun@vmware.com>

### CODE REVIEW

File: src/portal/src/app/project/repository/artifact-list-page/artifact-list/artifact-list-tab/artifact-list-tab.component.spec.ts [hunk 1]
```diff
@@ -359,17 +358,16 @@ describe("ArtifactListTabComponent (inline template)", () => {
     fixture.detectChanges();
   });
 
-  it("should load and render data", () => {
+  it("should load and render data", async () => {
```
Comment (selina-cui) [inline src/portal/src/app/project/repository/artifact-list-page/artifact-list/artifact-list-tab/artifact-list-tab.component.spec.ts:361]: waitForAsync ？

File: src/portal/src/app/shared/route/member-guard-activate.service.spec.ts [hunk 2]
```diff
@@ -1,4 +1,4 @@
-import { TestBed, async, inject } from '@angular/core/testing';
+import { TestBed, waitForAsync, inject } from '@angular/core/testing';
```
Comment (selina-cui) [inline src/portal/src/app/shared/route/member-guard-activate.service.spec.ts:1]: Is waitForAsync used in this file?

File: src/portal/src/app/shared/route/member-permission-guard-activate.service.spec.ts [hunk 3]
```diff
@@ -1,4 +1,4 @@
-import { TestBed, async, inject } from '@angular/core/testing';
+import { TestBed, waitForAsync, inject } from '@angular/core/testing';
```
Comment (selina-cui) [inline src/portal/src/app/shared/route/member-permission-guard-activate.service.spec.ts:1]: Is waitForAsync used in this file?

File: src/portal/src/app/shared/route/mode-guard-activate.service.spec.ts [hunk 4]
```diff
@@ -1,4 +1,4 @@
-import { TestBed, async, inject } from '@angular/core/testing';
+import { TestBed, waitForAsync, inject } from '@angular/core/testing';
```
Comment (selina-cui) [inline src/portal/src/app/shared/route/mode-guard-activate.service.spec.ts:1]: Is waitForAsync used in this file?

File: src/portal/src/app/shared/route/oidc-guard-active.service.spec.ts [hunk 5]
```diff
@@ -1,4 +1,4 @@
-import { TestBed, async, inject } from '@angular/core/testing';
+import { TestBed, waitForAsync, inject } from '@angular/core/testing';
```
Comment (selina-cui) [inline src/portal/src/app/shared/route/oidc-guard-active.service.spec.ts:1]: Is waitForAsync used in this file?

File: src/portal/src/app/shared/route/sign-in-guard-activate.service.spec.ts [hunk 6]
```diff
@@ -1,4 +1,4 @@
-import { TestBed, async, inject } from '@angular/core/testing';
+import { TestBed, waitForAsync, inject } from '@angular/core/testing';
```
Comment (selina-cui) [inline src/portal/src/app/shared/route/sign-in-guard-activate.service.spec.ts:1]: Is waitForAsync used in this file?

File: src/portal/src/app/gc-page/gc-page.component.html [hunk 7]
```diff
@@ -1,12 +1,12 @@
...
     <div class="col-lg-12 col-md-12 col-sm-12 col-xs-12">
-        <h2 class="custom-h2 gc-title">{{'CONFIG.GC' | translate }}<span class="spinner spinner-inline ml-1 v-mid" [hidden]="!inProgress"></span></h2>
+        <h2 class="custom-h2 gc-title">{{'CONFIG.GC' | translate }}<span *ngIf="inProgress" class="spinner spinner-inline ml-1 v-mid"></span></h2>
         <clr-tabs>
             <clr-tab *ngIf="hasAdminRole">
                 <button id="config-gc" clrTabLink>{{'CONFIG.GC' | translate }}</button>
                 <ng-template [(clrIfActive)]="gcActive">
                     <clr-tab-content id="gc" *ngIf="hasAdminRole">
-                        <gc-config (loadingGcStatus)="getGcStatus($event)"></gc-config>
+                        <gc-config #cgCom (loadingGcStatus)="getGcStatus($event)"></gc-config>
```
Comment (selina-cui) [inline src/portal/src/app/gc-page/gc-page.component.html:9]: what's cgCom

File: src/portal/angular.json [hunk 8]
```diff
@@ -40,9 +40,10 @@
               }
```
Comment (jwangyangls) [inline src/portal/angular.json:40]: image
Is this alignment problem caused by the upgrade?

Comment (jwangyangls) [reply src/portal/angular.json:40]: image

File: src/portal/angular.json [hunk 9]
```diff
@@ -40,9 +40,10 @@
...
             ],
             "scripts": [
+              "node_modules/@webcomponents/webcomponentsjs/custom-elements-es5-adapter.js",
```
Comment (jwangyangls) [inline src/portal/angular.json:43]: image

File: src/portal/angular.json [hunk 10]
```diff
@@ -40,9 +40,10 @@
...
+              "node_modules/@webcomponents/webcomponentsjs/webcomponents-bundle.js",
```
Comment (jwangyangls) [inline src/portal/angular.json:44]: image

File: src/portal/angular.json [hunk 11]
```diff
@@ -40,9 +40,10 @@
...
               "node_modules/core-js/client/shim.min.js",
```
Comment (jwangyangls) [inline src/portal/angular.json:45]: image

File: src/portal/src/app/config/scanner/scanner-metadata/scanner-metadata.component.spec.ts [hunk 12]
```diff
@@ -1,4 +1,4 @@
-import { async, ComponentFixture, TestBed } from '@angular/core/testing';
+import { waitForAsync, ComponentFixture, TestBed } from '@angular/core/testing';
```
[the remark this answers is not in the document]
Comment (AllForNothing) [reply src/portal/src/app/config/scanner/scanner-metadata/scanner-metadata.component.spec.ts:1]: removed

File: src/portal/src/app/project/repository/artifact-list-page/artifact-list/artifact-list-tab/artifact-list-tab.component.spec.ts (hunk 1 above)
Comment (AllForNothing) [reply src/portal/src/app/project/repository/artifact-list-page/artifact-list/artifact-list-tab/artifact-list-tab.component.spec.ts:361]: This is  "async  and await", "waitForAsync()" replaces "async()"

File: src/portal/src/app/shared/route/member-guard-activate.service.spec.ts (hunk 2 above)
Comment (AllForNothing) [reply src/portal/src/app/shared/route/member-guard-activate.service.spec.ts:1]: removed

File: src/portal/src/app/shared/route/member-permission-guard-activate.service.spec.ts (hunk 3 above)
Comment (AllForNothing) [reply src/portal/src/app/shared/route/member-permission-guard-activate.service.spec.ts:1]: removed

File: src/portal/src/app/shared/route/mode-guard-activate.service.spec.ts (hunk 4 above)
Comment (AllForNothing) [reply src/portal/src/app/shared/route/mode-guard-activate.service.spec.ts:1]: removed

File: src/portal/src/app/shared/route/oidc-guard-active.service.spec.ts (hunk 5 above)
Comment (AllForNothing) [reply src/portal/src/app/shared/route/oidc-guard-active.service.spec.ts:1]: removed

File: src/portal/src/app/shared/route/sign-in-guard-activate.service.spec.ts (hunk 6 above)
Comment (AllForNothing) [reply src/portal/src/app/shared/route/sign-in-guard-activate.service.spec.ts:1]: removed

File: src/portal/src/app/gc-page/gc-page.component.html (hunk 7 above)
Comment (AllForNothing) [reply src/portal/src/app/gc-page/gc-page.component.html:9]: removed

File: src/portal/angular.json (hunk 8 above)
Comment (AllForNothing) [reply src/portal/angular.json:40]: Fixed

File: src/portal/angular.json (hunk 9 above)
Comment (AllForNothing) [reply src/portal/angular.json:43]: Fixed

File: src/portal/angular.json (hunk 11 above)
Comment (AllForNothing) [reply src/portal/angular.json:45]: Fixed

File: src/portal/angular.json (hunk 10 above)
Comment (AllForNothing) [reply src/portal/angular.json:44]: Fixed

**Review by selina-cui — APPROVED** [review]

**Review by ninjadq — APPROVED** [review]
lgtm

**Pull request merged**